src/kdtree.rs

```diff
@@ -7,7 +7,7 @@ use crate::util;
 
 #[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
 #[derive(Clone, Debug)]
-pub struct KdTree<A, T, U: AsRef<[A]>> {
+pub struct KdTree<A, T: std::cmp::PartialEq, U: AsRef<[A]>+ std::cmp::PartialEq> {
```

Comment:
would this be a breaking change?

Reply:
I'd have to see if anyone was using custom structs that don't derive partialeq. If so, then yes this would be breaking.

Reply:
then it's a breaking change and we'll bump the version accordingly

it also introduces the notion that duplicates(by coordinates or value) are kept upon addition to the tree